client/state/jetpack-connect/test/selectors.js

```diff
@@ -6,49 +6,257 @@ import { expect } from 'chai';
 /**
  * Internal dependencies
  */
-import { isCalypsoStartedConnection, getFlowType, getJetpackSiteByUrl, hasXmlrpcError } from '../selectors';
-
-const stateHasXmlrpcError = {
-	jetpackConnect: {
-		jetpackConnectAuthorize: {
-			authorizeError: {
-				message: 'transport error - HTTP status code was not 200 (502)'
-			},
-			authorizationCode: 'xxxx'
-		}
-	}
-};
-
-const stateHasNoError = {
-	jetpackConnect: {
-		jetpackConnectAuthorize: {
-			authorizeError: false
-		}
-	}
-};
-
-const stateHasNoAuthorizationCode = {
-	jetpackConnect: {
-		jetpackConnectAuthorize: {
-			authorizeError: {
-				message: 'Could not verify your request.'
-			}
-		}
-	}
-};
-
-const stateHasOtherError = {
-	jetpackConnect: {
-		jetpackConnectAuthorize: {
-			authorizeError: {
-				message: 'Jetpack: [already_connected] User already connected.'
-			},
-			authorizationCode: 'xxxx'
-		}
-	}
-};
+import {
+	getConnectingSite,
+	getAuthorize,
+	getAuthorizeQuery,
+	getAuthorizeSite,
+	getSessions,
+	getSSOSessions,
+	getSSO,
+	isCalypsoStartedConnection,
+	getFlowType,
+	getJetpackSiteByUrl,
+	hasXmlrpcError
+} from '../selectors';
 
 describe( 'selectors', () => {
+	describe( '#getConnectingSite()', () => {
+		it( 'should return undefined if user has not started connecting a site', () => {
+			const state = {
+				jetpackConnect: {}
+			};
+
+			expect( getConnectingSite( state ) ).to.be.undefined;
+		} );
+
+		it( 'should return the current connecting site if there is such', () => {
+			const jetpackConnectSite = {
+				url: '',
+				isFetching: true,
+				isFetched: false,
+				isRedirecting: false,
+				installConfirmedByUser: true,
+				isDismissed: false,
+				data: {
+					isDotCom: false,
+					notExists: false,
+					notWordPress: false,
+					notJetpack: false,
+					jetpackVersion: '4.3.1',
+					outdatedJetpack: false,
+					notActiveJetpack: false,
+					notConnectedJetpack: true,
+					alreadyOwned: true,
+					alreadyConnected: false
+				}
+			};
+			const state = {
+				jetpackConnect: {
+					jetpackConnectSite
+				}
+			};
+
+			expect( getConnectingSite( state ) ).to.eql( jetpackConnectSite );
```

Comment:
oh, but if your selectors return a copy of the state, if you want to modify anything there directly from the code, you need to do it on purpose (passing the state, modifying it, not using redux anymore). The problem here is that if you return a pointer to the real state tree, we could modify the state by mistake, as a side effect. It's too easy to just assume that when you do 
` const site = getConnectingSite( state ) `

you are creating a local variable, so you can just do  `site.url = null` without any repercusion. And you would be changing the state directly